### PR TITLE
ch4: fixes the completion notification for send in `ofi`

### PR DESCRIPTION
**Pull request by thomasgillis**
when completing the ofi event we have to decrement the completion_notification to have proper completion notification mechanism

## Author Checklist
* [ ] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [ ] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [ ] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: src/mpid/ch4/netmod/ofi/ofi_events.h
Comment (hzhou) [inline src/mpid/ch4/netmod/ofi/ofi_events.h:40]: Can we free the buffer and then simply call `MPID_Request_complete`?

**Comment by thomasgillis** [comment]
test:mpich/ch4/most
test:mpich/warnings

**Comment by thomasgillis** [comment]
test:mpich/ch4/most
test:mpich/warnings

**Comment by thomasgillis** [comment]
test:mpich/ch4/most
test:mpich/warnings

**Review by hzhou — APPROVED** [review]
LGTM. I am approving it. Make sure clear all tests before merging.

File: src/mpid/ch4/netmod/ofi/ofi_events.h
Comment (hzhou) [inline src/mpid/ch4/netmod/ofi/ofi_events.h:45]: Remove the extra comment.

**Comment by thomasgillis** [comment]
test:mpich/ch4/most
test:mpich/warnings

**Comment by hzhou** [comment]
UCX failures are unrelated (can be ignored in this PR). They are tracked in #6636 and #6641. Need fix the ofi failures.

File: src/mpid/ch4/netmod/ofi/ofi_events.c
Comment (hzhou) [inline src/mpid/ch4/netmod/ofi/ofi_events.c:133]: `mpi_errno` is uninitialized. You may simply remove `mpi_errno` and return MPI_SUCCESS here.

**Comment by thomasgillis** [comment]
test:mpich/ch4/most
test:mpich/warnings

**Review by raffenet — APPROVED** [review]
LGTM if tests are clean.

**Pull request merged**
